### ISSUE TITLE
Tag ordering sometimes is not displayed in the order it is declared when using the --tag command in the Cypress Cloud

### ISSUE THREAD

**Issue by laerteneto** [cypress-io/cypress#28970]
### Current behavior

The tags, sometimes, are not being displayed over the Cypress Cloud in the order we declare in the --tag command.

### Desired behavior

The tags should be displayed in the order we declared in the --tag command as the documentation implies so: 
https://docs.cypress.io/guides/guides/command-line#cypress-run-tag-lt-tag-gt

### Test code to reproduce

- Run any tests in the Cypress Cloud with "--tag '1,2,3,4,5,6,7,8,9,10,11,12,13,14,15'"
- Observe the tags are correctly being displayed in the order above in the Cypress Cloud
- Now, run any tests again with "--tag '1,2,3,4,5,testing,hello,hello2,hello3,hello4,hello5,hello6'"
- For some reason, you will be able to see that the order above is not the one displayed in the Cypress Cloud,

![image](https://github.com/cypress-io/cypress/assets/9399511/da1104d8-051b-4fed-aa75-d681801b1143)

### Cypress Version

13.6.1

### Node version

20.6.1

### Operating System

Windows 10

**Comment by jennifer-shehane** [comment]
@laerteneto Hi, can you explain the importance of having the tags display in the order you defined them? How will this help your workflow? It will help understanding this in order to prioritize the work. Thanks!

**Comment by laerteneto** [comment]
@jennifer-shehane Hi, thanks for the reply.

We have lots of tags in our Cloud, such as branch names, app code versions from front-end and back-end, environment IDs, jenkins builds number, and some others. So, in our case, as we have many daily runs in the Cloud with many different sets of tags, people inside the company get a bit confused while taking an overview of all the runs from the Latest Runs page as the tags are not sorted in the way we declared to make the visualization of the runs easier. 
Basically, in my personal experience, this is something that can make people very confused while working with so many daily runs and with many tags with many differents set of tags in each run as sometimes.

To overcome this situation a little bit, we use the filters. However, it still does not solve the issue for us as the tags are still not sorted in the way we declared in the code sometimes, it only decreases the number of data retrieved, which is helpful, but it still makes people a bit confused.

**Comment by jennifer-shehane** [comment]
@laerteneto Thanks for taking the time to explain. I've forwarded your request to our Cloud team for them to evaluate the priority of this request.

**Comment by ruben0251** [comment]
@jennifer-shehane Will tag ordering be added in the future? We are struggling with the same issue.

**Comment by laerteneto** [comment]
Hello @jennifer-shehane do you know if there is any updates on this? Thanks
